recipes/gmp/all/conanfile.py

```diff
@@ -100,37 +96,37 @@ def generate(self):
         tc = AutotoolsToolchain(self)
         yes_no = lambda v: "yes" if v else "no"
         tc.configure_args.extend([
-            "--with-pic={}".format(yes_no(self.options.get_safe("fPIC", True))),
-            "--enable-assembly={}".format(yes_no(not self.options.get_safe("disable_assembly", False))),
-            "--enable-fat={}".format(yes_no(self.options.get_safe("enable_fat", False))),
-            "--enable-cxx={}".format(yes_no(self.options.enable_cxx)),
-            "--srcdir={}".format(self.source_folder.replace("\\", "/")),
+            f'--with-pic={yes_no(self.options.get_safe("fPIC", True))}',
+            f'--enable-assembly={yes_no(not self.options.get_safe("disable_assembly", False))}',
+            f'--enable-fat={yes_no(self.options.get_safe("enable_fat", False))}',
+            f'--enable-cxx={yes_no(self.options.enable_cxx)}',
+            f'--srcdir={"../src"}', # Use relative path to avoid issues with #include "$srcdir/gmp-h.in" on Windows
         ])
+        env = tc.environment()
         if is_msvc(self):
             tc.configure_args.extend([
                 "ac_cv_c_restrict=restrict",
                 "gmp_cv_asm_label_suffix=:",
                 "lt_cv_sys_global_symbol_pipe=cat",  # added to get further in shared MSVC build, but it gets stuck later
             ])
             tc.extra_cxxflags.append("-EHsc")
-            if (str(self.settings.compiler) == "Visual Studio" and Version(self.settings.compiler.version) >= "12") or \
-               (str(self.settings.compiler) == "msvc" and Version(self.settings.compiler.version) >= "180"):
+            if (self.settings.compiler == "msvc" and Version(self.settings.compiler.version) >= "180") or \
+               (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) >= "12"):
                 tc.extra_cflags.append("-FS")
                 tc.extra_cxxflags.append("-FS")
-        env = tc.environment()
-        if is_msvc(self):
+
             yasm_wrapper = unix_path(self, os.path.join(self.source_folder, "yasm_wrapper.sh"))
             yasm_machine = {
                 "x86": "x86",
                 "x86_64": "amd64",
             }[str(self.settings.arch)]
-            ar_wrapper = unix_path(self, self._user_info_build["automake"].ar_lib)
+            ar_wrapper = unix_path(self, self.conf.get("user.automake:lib-wrapper"))
             dumpbin_nm = unix_path(self, os.path.join(self.source_folder, "dumpbin_nm.py"))
             env.define("CC", "cl -nologo")
             env.define("CCAS", f"{yasm_wrapper} -a x86 -m {yasm_machine} -p gas -r raw -f win32 -g null -X gnu")
             env.define("CXX", "cl -nologo")
             env.define("LD", "link -nologo")
-            env.define("AR", f"{ar_wrapper} \"lib -nologo\"")
+            env.define("AR", f'{ar_wrapper} "lib -nologo"')
```

Comment:
keep these extra quotes please, it may fail without them

Reply:
I believe these are equivalent. As I understand it, the backslashes only affect Python. I try to always use single quotes with f-strings so that it is safe to use double quotes inside them.

Reply:
Indeed it's equivalent. I thought that you were removing double quotes around lib -nologo. I prefer previous syntax I've proposed, but why not.